### ISSUE TITLE
cleanup: enable `gocritic` linter

### ISSUE THREAD

**Issue by vincepri** [kubernetes-sigs/cluster-api-provider-aws#1543]
After #1535, we updated the golangci-lint to the latest version and matched the upstream Cluster API format for the linter, it seems there were a bunch of linters missing.

- Head over to [.golanci-lint.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/.golangci.yml) in the root of this repository.
- Remove the linter from the `linters.disable` list.
- Run `make lint-full`.
- Fix the reported errors with the suggestions.
- Submit a PR with the changes.

Feel free to reach out at any time if you need guidance or help.

/kind cleanup
/help
/good-first-issue
/milestone Next
/priority backlog

**Comment by ericsmalling** [comment]
/assign

I'll take a crack at this

**Comment by detiber** [comment]
@ericsmalling thanks!
/lifecycle active

**Comment by ericsmalling** [comment]
@detiber While looking into this, I ran into comments at https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/0f7e2dbd4204c981beeda4d5491bb4ff564cad1b/pkg/cloud/services/ec2/instances.go#L158 that mention a desire to refactor into its own func/method.  I agree with that assessment but it doesn't appear there are tests covering much of that code block (seems like just happy path covered)

This functional decomposition and adding of test cases seems out of scope for this issue, do you agree or should I add to this issue's PR with such changes?  If no, I can open another issue to do that work.

**Comment by detiber** [comment]
@ericsmalling +1 to creating a separate issue/pr to do the refactoring and adding test cases there.
